.github/bin/common.sh

```diff
@@ -164,39 +164,91 @@ installYq() {
   set +e
-  chectl server:logs --chenamespace=${NAMESPACE} --directory=${ARTIFACTS_DIR}
-  collectDevworkspaceOperatorLogs
-  oc get events -n ${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE}.txt
-  oc get events -n ${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE}.txt
+  collectClusterData
   set -e
 }
 
-collectDevworkspaceOperatorLogs() {
-  mkdir -p ${ARTIFACTS_DIR}/devworkspace-operator
-
-  oc get events -n devworkspace-controller > ${ARTIFACTS_DIR}/events-devworkspace-controller.txt
+collectClusterData() {
+  allNamespaces=$(kubectl get namespaces -o custom-columns=":metadata.name")
+  for namespace in $allNamespaces ; do
+    collectK8sResourcesForNamespace $namespace
+    collectPodsLogsForNamespace $namespace
+    collectEventsForNamespace $namespace
+  done
+  collectClusterScopeK8sResources
+}
+
+collectK8sResourcesForNamespace() {
+  namespace="$1"
+  if [[ -z $namespace ]]; then return; fi
+
+  STANDARD_KINDS=("pods" "jobs" "deployments"
+                  "services" "ingresses"
+                  "configmaps" "secrets"
+                  "serviceaccounts" "roles" "rolebindings"
+                  "pv" "pvc"
+                 )
+  CRDS_KINDS=($(kubectl get crds -o jsonpath="{.items[*].spec.names.plural}"))
+  KINDS=("${STANDARD_KINDS[@]}" "${CRDS_KINDS[@]}")
+
+  for kind in "${KINDS[@]}" ; do
+    dir="${ARTIFACTS_DIR}/resources/namespaced/${namespace}/${kind}"
+    mkdir -p $dir
+
+    names=$(kubectl get -n $namespace $kind --no-headers=true -o custom-columns=":metadata.name")
+    for name in $names ; do
+      name=${name//[:<>|*?]/_}
+      kubectl get -n $namespace $kind $name -o yaml > "${dir}/${name}.yaml"
+    done
+  done
+}
 
-  #determine the name of the devworkspace controller manager pod
-  local CONTROLLER_POD_NAME=$(oc get pods -n devworkspace-controller -l app.kubernetes.io/name=devworkspace-controller -o json | jq -r '.items[0].metadata.name')
-  local WEBHOOK_SVR_POD_NAME=$(oc get pods -n devworkspace-controller -l app.kubernetes.io/name=devworkspace-webhook-server -o json | jq -r '.items[0].metadata.name')
+collectClusterScopeK8sResources() {
+  declare -a KINDS=("crds"
+                    "clusterroles" "clusterrolebindings"
+                   )
+  for kind in "${KINDS[@]}" ; do
+    dir="${ARTIFACTS_DIR}/resources/cluster/${kind}"
+    mkdir -p $dir
 
-  # save the logs of all the containers in the DWO pod
-  for container in $(oc get pod -n devworkspace-controller ${CONTROLLER_POD_NAME} -o json | jq -r '.spec.containers[] | .name'); do
-    mkdir -p ${ARTIFACTS_DIR}/devworkspace-operator/${CONTROLLER_POD_NAME}
-    oc logs -n devworkspace-controller deployment/devworkspace-controller-manager -c ${container} > ${ARTIFACTS_DIR}/devworkspace-operator/${CONTROLLER_POD_NAME}/${container}.log
+    names=$(kubectl get -n $namespace $kind --no-headers=true -o custom-columns=":metadata.name")
```

Comment:
If that's a cluster scope objects, why do we need namespace argument?

Reply:
This function is designed for namespaced objects only. There is a separate one for luster scope objects.